pkg/gameserversets/gameserversets.go

```diff
@@ -65,6 +67,7 @@ func sortGameServersByLeastFullNodes(list []*agonesv1.GameServer, count map[stri
 			return false
 		}
 
+		// TODO: What if they're not on the same node -- do we want to delete the gameserver of "equal" Nodes where one Node has creating gameservers and the other has Ready gameservers?
 		// if both are in the same node, make sure to delete pre-Ready GameServers first
 		if a.Status.NodeName == b.Status.NodeName {
```

Comment:
(Assuming packed), if they are not in the same node, then we delete as per packed node rules (so least full first) -- which is what we have now.

Reply:
I'm wondering about states like:

a.Status.NodeName != b.Status.NodeName
acTotal == bcTotal && a.IsBeforeReady()?

Also wondering about:
a.IsBeforeReady() && b.Status.State == agonesv1.GameServerStateAllocated

Reply:
> a.Status.NodeName != b.Status.NodeName

That's handled by the node count sorting below.

> acTotal == bcTotal && a.IsBeforeReady()?

I don't think this happens, because state checks happen before node count checks.

> a.IsBeforeReady() && b.Status.State == agonesv1.GameServerStateAllocated

Not a concern for this function. Allocated state gets filtered later on.